lighthouse-core/test/audits/audit-test.js

```diff
@@ -37,7 +37,58 @@ describe('Audit', () => {
     assert.doesNotThrow(_ => B.audit());
   });
 
-  it('throws if an audit does return a result with a rawValue', () => {
-    assert.throws(_ => Audit.generateAuditResult(A, {}));
+  describe('_normalizeAuditScore', () => {
+    it('returns a score that is always 0-1', () => {
+      const auditResult = Audit._normalizeAuditScore(B, {rawValue: true});
+      assert.equal(Number.isFinite(auditResult.score), true);
+      assert.equal(auditResult.score, 1);
+      assert.equal(auditResult.score <= 1, true);
+
+      const auditResultFail = Audit._normalizeAuditScore(B, {rawValue: false});
+      assert.equal(Number.isFinite(auditResultFail.score), true);
+      assert.equal(auditResultFail.score, 0);
+      assert.equal(auditResultFail.score <= 1, true);
+      assert.equal(auditResultFail.score >= 0, true);
+    });
+
+    it('throws if an audit returns a score >1', () => {
+      assert.throws(_ => Audit._normalizeAuditScore(B, {rawValue: true, score: 100}), /is > 1/);
+      assert.throws(_ => Audit._normalizeAuditScore(B, {rawValue: true, score: 2}), /is > 1/);
+    });
+
+    it('throws if an audit returns a score that\'s not a number', () => {
+      const re = /Invalid score/;
+      assert.throws(_ => Audit._normalizeAuditScore(B, {rawValue: true, score: NaN}), re);
+      assert.throws(_ => Audit._normalizeAuditScore(B, {rawValue: true, score: '50'}), /is > 1/);
```

Comment:
wait why isn't this /Invalid score/?

Reply:
this pair of assertions: 
https://github.com/GoogleChrome/lighthouse/blob/scoring2.0/lighthouse-core/audits/audit.js#L127-L128

we assert its finite first, throw invalid score. otherwise we doublecheck we're finite but <= 1

Reply:
yeah exactly so shouldn't we throw invalid score... I'm so confused haha

![image](https://user-images.githubusercontent.com/2301202/37365212-145efb08-26ba-11e8-840d-0ba59810937f.png)

Reply:
yeah exactly. :D clamp casts. 
i fixed the tests. 

so right now one could pass in a score of `'0.7'` and it'd get casted and be just fine. we could hypothetically throw but yeah i don't think its a big deal.

i fixed the tests.